pkg/api/vlabs/validate.go

```diff
@@ -347,6 +347,11 @@ func (a *Properties) validateAgentPoolProfiles() error {
 		if e := validatePoolOSType(agentPoolProfile.OSType); e != nil {
 			return e
 		}
+		if agentPoolProfile.AcceleratedNetworkingEnabled != nil {
```

Comment:
Here you could use `helpers.IsTrueBoolPointer(agentPoolProfile.AcceleratedNetworkingEnabled)` instead of checking if it's nil and check helpers.AcceleratedNetworkingSupported(VMSize)` in validatePoolAcceleratedNetworking, no need to check if it is true or if AcceleratedNetworkingSupported == AcceleratedNetworking.